pitch.md

```diff
@@ -1,44 +1,160 @@
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+As mentioned above, in a perfect scenario the user would interact with their calendar
+in the most intuitive way for them. If thats with a UI, they should be able to easily
+create and edit the DSL with one. If they prefer typing, then the language should be
+comprehensible enough for humans to create and understand. Allowing user's access
+to the raw DSL also allows for the easy possibility of creating templates. Since
+everyone at a school generally has the same school year dates, with the same times
+for finals and other things, it would be able to save that as a file and send it to 
+others. Another plus for being able to send scheduels around is for members of the 
+same teams at their jobs and people taking almost the exact same classes (core). 
+
+First, a text based language will be the focus but if time permits a GUI will be
+looked into. 
+
+For the text based language, the verbs and nouns should be similar to how current
+calendar programs are designed. I want it to be easy for the user to use and
+understand the language whether it be the first or hundreths time they look at it. 
+
+I want a calendar to be sectioned off, so that different events can be grouped 
+together. Sections can have an endless number of sub-sections which each 
+sub-section following all of the rules of its parent. Each section also has rules
+which are very important. Rules can denote times that the events in the group
+are active and times that they are inactive. 
+
+The other main part to the language are of course events. Events have times of the
+day and week they are active. These are denoted with rules also. Rules go even
+deeper for events though as they allow for repetition, importance levels
+and even being conditional on other events occuring at the same time or on the 
+same day. 
+
+This all culminates in a tree like system with each terminal node being an 
+event and each branching node being a section. 
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+This will describe the text based language. Since the possible UI extension 
+would just be a visual representation of the text based language, its 
+semantics should be basically the same. 
+
+When the program runs, it will take in the file containing the DSL and evauate
+it. It will return a static calendar for now, even though allowing dynamic 
+changes in an online system could be a stretch goal. It will compute all of 
+the sections, events and rules and return a calendar that follows all of them
+over a given time span. One obvious error is events that occur at the same
+time. For now, I am thinking that it will allow it, showing them just as
+overlapping. Maybe later I will realize that we will need to make the user
```

Comment:
I often schedule events that overlap slightly and plan on either leaving early or arriving late, or I schedule flex time that I don't actually expect to need.  I'd be unhappy if it didn't allow at all for overlapping.